File: src/main/java/org/yggdrasil/core/ledger/transaction/TransactionOutput.java
package org.yggdrasil.core.ledger.transaction;

import org.yggdrasil.core.utils.CryptoHasher;
import org.yggdrasil.core.utils.CryptoKeyGenerator;

import java.io.Serializable;
import java.math.BigDecimal;
import java.security.*;

public class TransactionOutput implements Serializable {

    // public key of the wallet in question
    // receiver must be able to sign an output in order
    // claim it.
    protected final byte[] address;
    protected final BigDecimal value;

    public TransactionOutput(byte[] address, BigDecimal value) {
        this.address = address;
        this.value = value;
    }

    public boolean isMine(PublicKey publicKey, byte[] signature) throws NoSuchAlgorithmException, InvalidKeyException, SignatureException {
        Signature verify = Signature.getInstance(CryptoKeyGenerator.getSignatureAlgorithm());
        verify.initVerify(publicKey);
        return CryptoHasher.isEqualHashes(this.address, CryptoHasher.generateWalletAddress(publicKey)) && verify.verify(signature);
    }

    public BigDecimal getValue() {
        return this.value;
    }

    @Override
    public String toString() {
        return String.format("TxOut(val=%d, address=%s)", value.intValue(), CryptoHasher.humanReadableHash(this.address));
    }

}

File: yggdrasil-core-csharp/yggdrasil-core/core/ledger/transaction/TransactionOutput.cs
using System.Security.Cryptography;

using yggdrasil_core.core.utils;

namespace yggdrasil_core.core.ledger.transaction
{
    public class TransactionOutput : LedgerHashableItem
    {
        private readonly byte[] _address;
        public byte[] Address { get { return _address; } }

        private readonly ulong _value;
        public ulong Value { get { return _value; } }

        public TransactionOutput(byte[] address, ulong value)
        {
            this._address = address;
            this._value = value;
        }

        public override string ToString()
        {
            return String.Format("TxOut(val={0}, address={1})", _value, CryptoHasher.ToHumanReadableHash(_address));
        }

        public bool IsMine(ECDiffieHellmanPublicKey publicKey, byte[] signature)
        {
            var verifier = ECDsa.Create(publicKey.ExportParameters());
            return verifier.VerifyHash(new byte[0], signature);
        }

        public byte[] Bytes()
        {
            byte[] bytes = new byte[0];
            bytes = (byte[])bytes.Concat(_address);
            
            using (MemoryStream ms = new MemoryStream())
            {
                using (BinaryWriter bw = new BinaryWriter(ms))
                {
                    bw.Write(_value);
                }
                bytes = (byte[])bytes.Concat(ms.ToArray());
                return bytes;
            }
        }
    }
}
